### ISSUE TITLE
MinTTY has suddenly stopped working for me as of version 3.1.5

### ISSUE THREAD

**Issue by dgeoffri** [mintty/mintty#1001]
I am using Windows 7 on a Core 2 Duo E8400.  I have never had issues previously, but after the latest update, trying to start it shows nothing and leaves a stackdump file in my home directory.  Version 3.1.4 and earlier are working as expected, but 3.1.5 and 3.1.6 do not start.  I'm happy to provide any other debug info if requested.

The stackdump file, if it's useful, contains only the following:
Stack trace:
Frame     Function  Args
0022C2FC  610239CC (0000018C, 0000EA60, 000000A4, 0022C36C)
0022C42C  6110247A (0022C44C, 002963D8, 002963D8, 002963D3)
0022C448  6112C582 (00000401, 000303E8, 00000000, 00000000)

**Comment by elieux** [comment]
If this is in MSYS2, read the [FAQ](https://www.msys2.org/wiki/FAQ/) entry no. 6; the issue has probably nothing to do with mintty.

**Comment by NamelessMofo** [comment]
Hi, I am having this exact same problem on Cygwin. I'm trying to update my portable Cygwin install that was admittedly very old (mintty version was 1.1.2!) and having this exact same issue. I'm trying to create both 32-bit and 64-bit portable installs, this issue cropping up with the 32-bit install, haven't tried 64-bit yet.

Doing "mintty -V" correctly pops up the version window, "mintty -H" correctly pops up the help window. Even just mintty.exe on a bash cmd line in Cygwin hits this stack dump issue.

**Comment by jay** [comment]
I updated my cygwin installation today and have this same problem. I was able to reproduce it in a virtual machine with a clean install. mintty 3.1.5 and 3.1.6 do not work. mintty 3.1.4 appears to work. Windows 7 Enterprise x64 and cygwin x86.

**Comment by dgeoffri** [comment]
I should have specified that this is on 32-bit Windows 7 Professional, and from cygwin, not MSYS2, and (it looks like current MSYS2 does not even have a 32-bit version)

**Comment by mintty** [comment]
@Biswa96, didn't you have an older Win 10 version at hand for testing?
Can you please check whether mintty 3.1.6 runs on it?

**Comment by Biswa96** [comment]
I could download it if we have internet here. There is no broadband connection due to cyclone.

**Comment by Biswa96** [comment]
I could download it if we have internet here. There is no broadband connection due to cyclone, only mobile internet which is slow as hell. Though I have Win7 and Win8.1 VMs. But I was wondering isn't the function pointer check from uxtheme.dll is sufficient to limit those in Win10?

**Comment by mintty** [comment]
The problem seems to be the undocumented and even unnamed functions employed in the solution and acquired via GetProcAddress(..., MAKEINTRESOURCEA(...)).
I've restored mintty to work in Windows 7 by guarding the whole thing as shown below and I'd like to check whether this is necessary for early Windows 10 versions as well.
```
  if (uxtheme) {
    DWORD win_version = GetVersion();
    win_version = ((win_version & 0xff) << 8) | ((win_version >> 8) & 0xff);
    if (win_version > 0x0601) {
      pShouldAppsUseDarkMode = 
        (void *)GetProcAddress(uxtheme, MAKEINTRESOURCEA(132)); /* ordinal */
      pSetPreferredAppMode = 
        (void *)GetProcAddress(uxtheme, MAKEINTRESOURCEA(135)); /* ordinal */
        // this would be AllowDarkModeForApp before Windows build 18362
    }

**Comment by NamelessMofo** [comment]
Thank you for the quick response on this bug! Will a release be forthcoming or should I just fall back to 3.1.4? Either solution works for me, honestly.

**Comment by mintty** [comment]
As mintty does not work in your environment right now, I'd assume you fallback to 3.1.4.
Next release will still need a few days, for another issue.

**Comment by Biswa96** [comment]
Or compile mintty from git repository (here) and place it in `/bin` folder in cygwin installation.

**Comment by mrodden** [comment]
Can confirm that I hit the same problem... 3.1.4 x86 (32 bit) works fine but 3.1.5 and 3.1.6 were broken for me. Running Win7 x86_64 with cygwin x86 (32bit)

**Comment by mintty** [comment]
The issue has likely been fixed (see commit above), for the next release.

**Comment by k-takata** [comment]
https://github.com/mintty/mintty/blob/d132f74f3582e7c1763ea0fce6d259bdd4370dd3/src/winmain.c#L248

I think this line should be:

```c
    if (win_version >= 0x0A00 && build >= 17763) { // minimum version 1809
```

**Comment by mintty** [comment]
Yes, thanks. Not critical as the build comparison should be sufficient to guard the version.

**Comment by mintty** [comment]
Released 3.1.7.
